src/Autofac/Builder/ReflectionActivatorData.cs

```diff
@@ -108,11 +108,11 @@ public IConstructorSelector ConstructorSelector
         /// <summary>
         /// Gets the explicitly bound constructor parameters.
         /// </summary>
-        public IList<Parameter> ConfiguredParameters { get; } = new List<Parameter>();
+        public List<Parameter> ConfiguredParameters { get; } = new List<Parameter>();
```

Comment:
I don't think this would cause any issues from the API perspective.

Reply:
Agreed, likely not a big deal. Unclear how much this stuff gets used, if ever, outside of here.

Reply:
I'm trying to put this together with the change from `Autofac.Core.Activators.Reflection.ReflectionActivator`. It looks like the change here wants to expose these as `List<T>`. These appear to be primarily used when instantiating a `ReflectionActivator`. The change in `ReflectionActivator` is to stop taking an `IEnumerable<T>` and take an `IList<T>` instead.

Could we leave this alone (as `IList<T>`) and _only_ make the change in `ReflectionActivator`? As it stands there's still going to be some implicit casting going on, which I'm guessing is the point of this.